ns/init_linux.go

```diff
@@ -76,11 +76,28 @@ func getSupportedNlFamilies() []int {
 }
 
 func loadXfrmModules() error {
-	if out, err := exec.Command("modprobe", "-va", "xfrm_user").CombinedOutput(); err != nil {
-		return fmt.Errorf("Running modprobe xfrm_user failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
+	if err := loadModule("xfrm_user"); err != nil {
+		return err
 	}
-	if out, err := exec.Command("modprobe", "-va", "xfrm_algo").CombinedOutput(); err != nil {
-		return fmt.Errorf("Running modprobe xfrm_algo failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
+	if err := loadModule("xfrm_algo"); err != nil {
+		return err
 	}
 	return nil
 }
+
+func loadModule(module string) error {
+	if !isModuleAlive(module) {
```

Comment:
Is this based on the assumption that the module is already loaded in the host before you started the daemon inside dind? Because of the module is not loaded you for sure cannot try to load it inside dind unless you bind mount certain host directories.